packages/space-opera/src/components/mobile_view/open_mobile_view.ts

```diff
@@ -195,19 +199,38 @@ export class OpenMobileView extends ConnectedLitElement {
     }
 
     if (updatedContent.envChanged && envBlob) {
-      await post(envBlob, envToSession(this.pipeId, session.id));
+      await post(
+          envBlob,
+          envToSession(this.pipeId, session.id, updatedContent.envIsHdr));
     }
 
     // The isStale flag will stay true if all of the requests are not delivered.
     session.isStale = false;
   }
 
+  async sendSessionContentHolder(
+      session: MobileSession, updatedContent: EditorUpdates,
+      usdzBlob: Blob|undefined, gltfBlob: Blob|undefined,
+      envBlob: Blob|undefined) {
+    try {
+      await this.sendSessionContent(
+          session, {...updatedContent}, usdzBlob, gltfBlob, envBlob);
+    } catch (e) {
+      console.log('error posting...');
+    }
+  }
+
   // Send any state, model, or environment iamge that has been updated since the
   // last refresh.
-  // TODO: Now that the posts are asynchronous, need to find a new way to deal
-  // with knowing when isSendingData should be set to false.
   async postInfo() {
+    if (this.isSendingData) {
+      return;
+    }
     this.isSendingData = true;
+    const sessionList = [...this.sessionList];
+    setTimeout(() => {
+      this.isSendingData = false;
+    }, 20000);
```

Comment:
Might be nice to replace times like these with `ALL_CAPS_NAMES` that are defined at the top of the file or imported from a parameters file, so that it's easy to know what they are and change them if necessary.